packages/core/src/lib/create-mapper/get-member-path.util.ts

```diff
@@ -78,26 +35,30 @@ export function getMembers(fnSelectorStr: string): string[] | null {
  * ```
  */
 export function getMemberPath(fn: Selector): string {
-  const fnString = fn
-    .toString()
-    // .replace(/\/\* istanbul ignore next \*\//g, '')
-    .replace(/cov_.+\n/g, '');
-
-  const cleaned = cleanseAssertionOperators(fnString);
-
   // Note that we don't need to remove the `return` keyword because, for instance,
   // `(x) => { return x.prop }` will be turn into `x=>returnx.prop` (cleaned)
   // thus we'll still be able to get only the string `prop` properly.
   // The same for `function (x){}`
-  const members = getMembers(cleaned);
+  const members = getMembers(fn);
 
   return members ? members.join('.') : '';
 }
 
 /**
- * @returns {string} The given `parseName` but without curly brackets, blank
- * spaces, semicolons, parentheses, "?" and "!" characters.
+ * @returns {Proxy} A proxy that's wrap on the target object and track of
+ * the path of accessed nested properties
  */
-function cleanseAssertionOperators(parsedName: string): string {
-  return parsedName.replace(/[\s{}()?!;]+/gm, '');
+// eslint-disable-next-line @typescript-eslint/ban-types
+function createProxy<T extends object>(target: T, path: string[] = []): T {
```

Comment:
nitpick: use `Dictionary<T>` instead of `object` here so we can remove the `eslint-disable` comment